conformance/tests/httproute-query-param-matching.go

```diff
@@ -84,6 +84,68 @@ var HTTPRouteQueryParamMatching = suite.ConformanceTest{
 			Response: http.Response{StatusCode: 404},
 		}}
 
+		// Combinations of query param matching with other core matches.
+		testCases = append(testCases, []http.ExpectedResponse{
+			{
+				Request:   http.Request{Path: "/path1?animal=whale"},
+				Backend:   "infra-backend-v1",
+				Namespace: ns,
+			},
+			{
+				Request:   http.Request{Headers: map[string]string{"version": "one"}, Path: "/?animal=whale"},
+				Backend:   "infra-backend-v2",
+				Namespace: ns,
+			},
+			{
+				Request:   http.Request{Headers: map[string]string{"version": "two"}, Path: "/path2?animal=whale"},
+				Backend:   "infra-backend-v3",
+				Namespace: ns,
+			},
+		}...)
+
+		// Ensure that combinations of matches which are OR'd together match
+		// even if only one of them is used in the request.
+		testCases = append(testCases, []http.ExpectedResponse{
+			{
+				Request:   http.Request{Path: "/path3?animal=shark"},
+				Backend:   "infra-backend-v1",
+				Namespace: ns,
+			},
+			{
+				Request:   http.Request{Headers: map[string]string{"version": "three"}, Path: "/path4?animal=kraken"},
+				Backend:   "infra-backend-v1",
+				Namespace: ns,
+			},
+		}...)
+
+		// Ensure that combinations of match types which are ANDed together do not match
+		// when only a subset of match types is used in the request.
+		testCases = append(testCases, []http.ExpectedResponse{
+			{
+				Request:  http.Request{Path: "/?animal=shark"},
+				Response: http.Response{StatusCode: 404},
+			},
+			{
+				Request:  http.Request{Path: "/path4?animal=kraken"},
+				Response: http.Response{StatusCode: 404},
+			},
+		}...)
+
+		// For requests that satisfy multiple matches, ensure precedence order
+		// defined by the Gateway API spec is maintained.
+		testCases = append(testCases, []http.ExpectedResponse{
+			{
+				Request:   http.Request{Path: "/path5?animal=hydra"},
+				Backend:   "infra-backend-v1",
+				Namespace: ns,
+			},
+			{
+				Request:   http.Request{Headers: map[string]string{"version": "four"}, Path: "/?animal=hydra"},
```

Comment:
Looks like this test case added after my review doesn't work with Contour

this assumes header matching gets precedence over method matching, but I'm not sure that is expressed anywhere explicitly in the spec, see https://github.com/kubernetes-sigs/gateway-api/issues/1993

Reply:
Yes that was how I interpreted it -- given method matching is not in the list (and the only thing not in the list), it must have the least priority.

I'm sorry for causing any issues. I've sent https://github.com/kubernetes-sigs/gateway-api/pull/2052 as a temporary solution to help unblock (if this is the path we wish to take)

Reply:
thanks! yeah not a super strong opinion on where it lands in terms of priority, but should probably be something explicitly debated/documented 👍🏽